### ISSUE TITLE
Validate blob belongs to the app when staging

### ISSUE THREAD

**Issue by jimmykarily** [epinio/epinio#801]
Depends on #699

When the user uploads the application code, we [store the application metadata](https://github.com/epinio/epinio/pull/788/files#diff-4c92e03256b6a12e87b841840188789dbcf00fcb04087875af0779129ac3cf3bR79-R81) on the object itself. We do the same [when we pull from git](https://github.com/epinio/epinio/pull/788/files#diff-676e4bf1f445a3687fbd3165180a5d030da0b4438d7a5ae481f67ebb758e4054R83-R85)

We can use these metadata to validate that the blob the client sends on the stage request actually matches the application on the same request. This should prevent someone from staging an app with the sources of another app.

Given everyone can do everything on Epinio right now (we don't have user roles), this is mostly for validation against code mistakes on the client (e.g. the web UI).

After discussing it with @manno, we think it could even be a feature to be able to deploy applications with the blob of another app. E.g. Someone updates the sources of a wordpress application and everyone else is staging their apps using those (updated) sources. To make it a useful feature, it would require a lot more that simply allowing this.

TBD: To validate or not to validate?

**Comment by agracey** [comment]
A bit of a tangent, but it would be nice if the validation added to the security of the platform in the future. Maybe PKI?

**Comment by jimmykarily** [comment]
> A bit of a tangent, but it would be nice if the validation added to the security of the platform in the future. Maybe PKI?

That's a good idea! So Epinio only uses blobs it uploaded and signed. This will prevent someone from messing with the uploaded blobs by accessing the S3 bucket directly. Maybe we can split that to a separate issue. Let's wait until this one is picked up to decide.

**Comment by andreas-kupries** [comment]
The referenced pieces of code are:

```
blobUID, err := manager.Upload(ctx, blob, map[string]string{
		"app": name, "org": org, "username": username,
	})
```
where the map contains the meta data in question.
